### PR TITLE
Migrate App Config tests to TokenCredential Auth

### PR DESCRIPTION
**Pull request by g2vinay**
Migrates App Config to use Token Credential auth in tests.

### CODE REVIEW

File: sdk/appconfiguration/tests.native.yml [hunk 1]
```diff
@@ -19,9 +19,5 @@ extends:
       safeName: azuredataappconfiguration
     TimeoutInMinutes: 90
     SupportedClouds: 'Public'
-    EnvVars:
```
Comment (danieljurek) [inline sdk/appconfiguration/tests.native.yml:22]: With no `UseFederatedAuth: true` these tests will not run with federated auth.

File: sdk/appconfiguration/test-resources.json [hunk 2]
```diff
@@ -1,9 +1,14 @@
+// cspell:ignore azconfig
```
Comment (danieljurek) [inline sdk/appconfiguration/test-resources.json:1]: Add this to `words` in `.vscode/cspell.json`

File: sdk/appconfiguration/tests.native.yml (hunk 1 above)
Comment (g2vinay) [reply sdk/appconfiguration/tests.native.yml:22]: added.

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java [hunk 3]
```diff
@@ -64,4 +69,14 @@ private static boolean shouldServiceVersionBeTested(ConfigurationServiceVersion
...
             serviceVersion.getVersion().equals(configuredServiceVersion.trim()));
     }
+
+    static TokenCredential getTokenCredential(InterceptorManager interceptorManager) {
+        if (interceptorManager.isPlaybackMode()) {
+            return new MockTokenCredential();
+        } else if (interceptorManager.isRecordMode()) {
+            return new DefaultAzureCredentialBuilder().build();
+        } else {
+            return new AzurePowerShellCredentialBuilder().build();
```
Comment (billwert) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java:79]: This won't work in our current beta; all the params moved to required params on the builder. Ex:
```
  AzurePipelinesCredential credential = new AzurePipelinesCredentialBuilder()
      .clientId(clientId)
      .tenantId(tenantId)
      .serviceConnectionId(serviceConnectionId)
      .systemAccessToken(systemAccessToken)
      .build();
 ```

Comment (g2vinay) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java:79]: This is using Powershell Credential here, not PipelinesCredential.

Comment (g2vinay) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java:79]: PipelinesCredential support will be added later on as we update our Pipelines config.

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java [hunk 4]
```diff
@@ -23,24 +22,21 @@
...
 
     private void setup(HttpClient httpClient, ConfigurationServiceVersion serviceVersion) {
         if (interceptorManager.isPlaybackMode()) {
-            connectionString = FAKE_CONNECTION_STRING;
-            String endpoint = new ConfigurationClientCredentials(connectionString).getBaseUri();
+            String endpoint = new ConfigurationClientCredentials(FAKE_CONNECTION_STRING).getBaseUri();
             // In playback mode use connection string because CI environment doesn't set up to support AAD
             client = new ConfigurationClientBuilder()
                 .credential(new MockTokenCredential())
                 .endpoint(endpoint)
```
Comment (billwert) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java:34]: Consider using `getTokenCredential` everywhere to reinforce the pattern

File: eng/pipelines/templates/steps/build-and-test.yml [hunk 5]
```diff
@@ -74,23 +80,53 @@ steps:
...
-      publishJUnitResults: false
-      goals: ${{ parameters.TestGoals }}
-    env: ${{ parameters.TestEnvVars }}
-    condition: and(succeeded(), ne(variables['TestFromSource'], 'true'))
+  - ${{ if eq('true', parameters.UseFederatedAuth) }}:
+    # Remove preceeding "1." in the version specified by $(JavaTestVersion)
+    # e.g. 1.8 -> 8
+    # This is required to set JAVA_HOME correctly for the Java version
+    - pwsh: |
+        $javaTestVersion = "$(JavaTestVersion)" -replace '^\d*?\.', ''
```
Comment (billwert) [inline eng/pipelines/templates/steps/build-and-test.yml:88]: this is fine but `"$(JavaTestVesrion)".Substring(2)` might be clearer? It's always 1. right?

File: eng/pipelines/templates/steps/build-and-test.yml [hunk 6]
```diff
@@ -74,23 +80,53 @@ steps:
...
+        displayName: '${{ parameters.VersionOverride }} - Run tests (Federated Auth)'
+      ${{ else }}:
+        displayName: 'Run tests (Federated Auth)'
+      inputs:
+        azureSubscription: ${{ parameters.ServiceConnection }}
+        azurePowerShellVersion: LatestVersion
+        pwsh: true
+        ScriptType: InlineScript
+        Inline: |
+          mvn --% -f $(Build.SourcesDirectory)/pom.xml $(DefaultTestOptions) -DAZURE_TEST_DEBUG=$(IsDebug) -Dparallel-test-playback=${{ parameters.ParallelTestPlayback }} ${{ parameters.TestOptions }} -T ${{ parameters.TestParallelization }} ${{ parameters.TestGoals }}
```
Comment (billwert) [inline eng/pipelines/templates/steps/build-and-test.yml:106]: Can we build the options once and use it in both places? This feels like a maintenance risk.

**Review by billwert — APPROVED** [review]
LGTM from a "what changes a test needs" perspective. I looked through the pipelines and it seems good to me, just left a few comments to consider.

**Review by mssfang — APPROVED** [review]
LGTM! Thanks Vinay

File: eng/pipelines/templates/steps/build-and-test.yml (hunk 6 above)
Comment (danieljurek) [reply eng/pipelines/templates/steps/build-and-test.yml:106]: We'll eventually move away from `AzurePowerShell` so this whole thing can be removed when we do that.

File: sdk/appconfiguration/test-resources.json (hunk 2 above)
Comment (danieljurek) [reply sdk/appconfiguration/test-resources.json:1]: Delete the comment, too

File: sdk/appconfiguration/test-resources.json [hunk 7]
```diff
@@ -57,9 +61,9 @@
         }
     ],
     "outputs": {
-        "AZURE_APPCONFIG_CONNECTION_STRING": {
-            "type": "string",
-            "value": "[listKeys(resourceId('Microsoft.AppConfiguration/configurationStores',variables('uniqueAzConfigName')), '2019-02-01-preview').value[0].connectionString]"
+        "AZ_CONFIG_ENDPOINT": {
```
Comment (samvaity) [inline sdk/appconfiguration/test-resources.json:64]:
```suggestion
        "AZ_APP_CONFIG_ENDPOINT": {
```

**Review by samvaity — APPROVED** [review]
Should the endpoint var name be `AZ_APP_CONFIG_ENDPOINT`. Other changes lgtm.

**Review by weshaggard — APPROVED** [review]

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java (hunk 3 above)
Comment (rujche) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java:79]: > PipelinesCredential support will be added later on as we update our Pipelines config.

@g2vinay Please let me when PipelinesCredential ready to use.

Comment (rujche) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java:79]: Is it possible to use `DefaultAzureCredential`, and let `DefaultAzureCredential` choose `AzurePowerShellCredentialBuilder` or `PipelinesCredential` automatically when run in pipeline?

**Pull request merged**